mongo/client_test.go

```diff
@@ -179,6 +179,50 @@ func TestClient(t *testing.T) {
 		client := setupClient(options.Client().SetReadConcern(rc))
 		assert.Equal(t, rc, client.readConcern, "expected read concern %v, got %v", rc, client.readConcern)
 	})
+	t.Run("min pool size", func(t *testing.T) {
```

Comment:
Nice tests! I think it's good to test the same functionality through `ApplyURI` and the `Set*` methods.